h/static/scripts/test/streamer-test.js

```diff
@@ -83,27 +83,66 @@ describe('streamer', function () {
   it('should not create a websocket connection if websocketUrl is not provided', function () {
     fakeSettings = {};
     createDefaultStreamer();
+    activeStreamer.connect();
+    assert.isNull(fakeWebSocket);
+  });
+
+  it('should not create a websocket connection if the user is not logged in', function () {
+    createDefaultStreamer();
     assert.isNull(fakeWebSocket);
   });
 
-  it('should send a client ID', function () {
+  it('should create a websocket connection if the user is logged in', function () {
+    fakeSession.state = {userid: 'foo'};
+    createDefaultStreamer();
+    assert.ok(fakeWebSocket);
+  });
+
+  it('should create a websocket connection if explicitly connected', function () {
+    createDefaultStreamer();
+    activeStreamer.connect();
+    assert.ok(fakeWebSocket);
+  });
+
+  it('should have a non-null client ID', function () {
     createDefaultStreamer();
+    assert.ok(activeStreamer.clientId);
+  });
+
+  it('should send the client ID on connection', function () {
+    createDefaultStreamer();
+    activeStreamer.connect();
     assert.equal(fakeWebSocket.messages.length, 1);
     assert.equal(fakeWebSocket.messages[0].messageType, 'client_id');
     assert.equal(fakeWebSocket.messages[0].value, activeStreamer.clientId);
   });
 
-  it('should close any existing socket', function () {
-    createDefaultStreamer();
-    var oldWebSocket = fakeWebSocket;
-    activeStreamer.connect();
-    assert.ok(oldWebSocket.didClose);
-    assert.ok(!fakeWebSocket.didClose);
+  describe('#connect()', function () {
+    it('should not close any existing socket', function () {
+      createDefaultStreamer();
+      activeStreamer.connect();
+      var oldWebSocket = fakeWebSocket;
+      activeStreamer.connect();
+      assert.ok(!oldWebSocket.didClose);
+      assert.ok(!fakeWebSocket.didClose);
+    });
+  });
+
+  describe('#reconnect()', function () {
```

Comment:
Yep, I forgot to change the `connect()` call there to `reconnect()` when the former became non-destructive. Fixed now.